dlc/src/main/scala/org/bitcoins/dlc/execution/DLCExecutor.scala

```diff
@@ -85,33 +85,74 @@ case class DLCExecutor(signer: DLCTxSigner)(implicit ec: ExecutionContext) {
     */
   def executeDLC(
       dlcSetup: SetupDLC,
-      oracleSigs: Vector[OracleSignatures]): Future[ExecutedDLCOutcome] = {
+      oracleSigs: Vector[OracleSignatures]): ExecutedDLCOutcome = {
+    val remoteFundingPubKey = if (isInitiator) {
+      builder.acceptFundingKey
+    } else {
+      builder.offerFundingKey
+    }
+
+    DLCExecutor.executeDLC(dlcSetup.cets,
+                           oracleSigs,
+                           signer.fundingKey,
+                           remoteFundingPubKey,
+                           builder.contractInfo,
+                           dlcSetup.fundingTx)
+  }
+
+  def executeRefundDLC(dlcSetup: SetupDLC): RefundDLCOutcome = {
+    val SetupDLC(fundingTx, _, refundTx) = dlcSetup
+    RefundDLCOutcome(fundingTx, refundTx)
+  }
+}
+
+object DLCExecutor {
 
-    val threshold = dlcSetup.cets.head._1.oracles.length
+  def executeDLC(
+      remoteCETInfos: Vector[(OracleOutcome, CETInfo)],
+      oracleSigs: Vector[OracleSignatures],
+      fundingKey: ECPrivateKey,
+      remoteFundingPubKey: ECPublicKey,
+      contractInfo: ContractInfo,
+      fundingTx: Transaction
+  ): ExecutedDLCOutcome = {
+    val threshold = contractInfo.oracleInfo.threshold
     val sigCombinations = CETCalculator.combinations(oracleSigs, threshold)
 
     var msgOpt: Option[OracleOutcome] = None
     val sigsUsedOpt = sigCombinations.find { sigs =>
-      msgOpt = builder.contractInfo.findOutcome(sigs)
+      msgOpt = contractInfo.findOutcome(sigs)
       msgOpt.isDefined
     }
-    val (msg, remoteAdaptorSig) = msgOpt match {
-      case Some(msg) =>
-        val cetInfo = dlcSetup.getCETInfo(msg)
-        (msg, cetInfo.remoteSignature)
+
+    val msgAndCETInfoOpt = msgOpt.flatMap { msg =>
+      remoteCETInfos.find(_._1 == msg)
+    }
+
+    val (msg, ucet, remoteAdaptorSig) = msgAndCETInfoOpt match {
+      case Some((msg, CETInfo(ucet, remoteSig))) => (msg, ucet, remoteSig)
       case None =>
         throw new IllegalArgumentException(
```

Comment:
so is this a "fatal" error? This seems really bad to me, an outcome occurred we didn't know about?

Reply:
I mean if all the oracles sign "hehe, screw my users" there isn't really anything you can do about that.